lib/ansible/module_utils/facts/network/linux.py

```diff
@@ -263,6 +263,13 @@ def parse_ip_output(output, secondary=False):
             parse_ip_output(primary_data)
             parse_ip_output(secondary_data, secondary=True)
 
+            if rc != 0:
+                # possibly busybox, fallback to running without the "primary" arg
+                # https://github.com/ansible/ansible/issues/50871
+                args = [ip_path, 'addr', 'show', device]
+                rc, data, stderr = self.module.run_command(args, errors='surrogate_then_replace')
```

Comment:
Check the `rc` of this command to make sure it was good before parsing. You should probably do this on the previous commands as well.

Reply:
:+1: